var/spack/repos/builtin/packages/pegtl/package.py

```diff
@@ -21,3 +21,6 @@ class Pegtl(CMakePackage):
     version('develop', branch='master')
     version('2.1.4', sha256='d990dccc07b4d9ba548326d11c5c5e34fa88b34fe113cb5377da03dda29f23f2')
     version('2.0.0', sha256='5aae0505077e051cae4d855c38049cc6cf71103a6cc8d0ddef01a576e8a60cc0')
+
+    # Ref: https://github.com/taocpp/PEGTL/blob/master/src/example/pegtl/json_classes.hpp
+    patch('change_to_virtual_destructor.patch')
```

Comment:
It looks like this was fixed in 2.5.0? In that case, add: `when='@:2.4'` so that it doesn't get applied to the `master` branch.